### PR TITLE
Yatopia fix: LinkedHashSet -> AbstractCollection

### LINKED ISSUE #72: Yatopia Monster Spawn Conflict
Hello!

As you can tell, I am running a server using "[Yatopia](https://github.com/YatopiaMC/Yatopia)" which is a fork of paper that includes many optimization patches, of their own, and many taken from other repos like Purpur and EMC. I'm sure this may cause issues with some plugins but so far I have had none, however I think this issue has yatopia to blame, they use a custom optimized linked hash set object.

```
[02:44:35 ERROR]: Could not pass event CreatureSpawnEvent to Vivecraft-Spigot-Extensions v1.16.3-r1
java.lang.ClassCastException: it.unimi.dsi.fastutil.objects.ObjectLinkedOpenHashSet cannot be cast to java.util.LinkedHashSet
at org.vivecraft.VSE.EditEntity(VSE.java:322) ~[?:?]
at org.vivecraft.VSE.onEvent(VSE.java:305) ~[?:?]
at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor230.execute(Unknown Source) ~[?:?]
at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at org.bukkit.craftbukkit.v1_16_R2.event.CraftEventFactory.callCreatureSpawnEvent(CraftEventFactory.java:673) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at org.bukkit.craftbukkit.v1_16_R2.event.CraftEventFactory.doEntityAddEventCalling(CraftEventFactory.java:594) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at net.minecraft.server.v1_16_R2.WorldServer.addEntity0(WorldServer.java:1678) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at net.minecraft.server.v1_16_R2.WorldServer.addEntity(WorldServer.java:1579) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at net.minecraft.server.v1_16_R2.WorldAccess.lambda$addAllEntities$0(WorldAccess.java:13) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:419) ~[?:1.8.0_242]
at java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:742) ~[?:1.8.0_242]
at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:647) ~[?:1.8.0_242]
at net.minecraft.server.v1_16_R2.WorldAccess.addAllEntities(WorldAccess.java:13) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at net.minecraft.server.v1_16_R2.SpawnerCreature.spawnMobsInternal(SpawnerCreature.java:261) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at net.minecraft.server.v1_16_R2.SpawnerCreature.spawnMobs(SpawnerCreature.java:187) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at net.minecraft.server.v1_16_R2.SpawnerCreature.a(SpawnerCreature.java:158) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at net.minecraft.server.v1_16_R2.ChunkProviderServer.tickChunks(ChunkProviderServer.java:1030) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at net.minecraft.server.v1_16_R2.ChunkProviderServer.tick(ChunkProviderServer.java:926) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at net.minecraft.server.v1_16_R2.WorldServer.doTick(WorldServer.java:887) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at net.minecraft.server.v1_16_R2.MinecraftServer.b(MinecraftServer.java:1434) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at net.minecraft.server.v1_16_R2.DedicatedServer.b(DedicatedServer.java:390) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at net.minecraft.server.v1_16_R2.MinecraftServer.a(MinecraftServer.java:1282) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at net.minecraft.server.v1_16_R2.MinecraftServer.w(MinecraftServer.java:1007) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at net.minecraft.server.v1_16_R2.MinecraftServer.lambda$a$0(MinecraftServer.java:165) ~[patched_1.16.3.jar:git-Yatopia-"0faeed7"]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_242]```

### PR DESCRIPTION
**Pull request by LittleBigBug**
Fixes #72

### CODE REVIEW

File: src/org/vivecraft/VSE.java [hunk 1]
```diff
@@ -7,12 +7,7 @@
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.net.URL;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.UUID;
+import java.util.*;
```
Comment (Techjar) [inline src/org/vivecraft/VSE.java:10]: Can we like not mess with all the imports?

**Review by Techjar — CHANGES_REQUESTED** [review]

File: src/org/vivecraft/VSE.java (hunk 1 above)
Comment (LittleBigBug) [reply src/org/vivecraft/VSE.java:10]: yes, sorry, I suspect my IDE did that.

Comment (Techjar) [reply src/org/vivecraft/VSE.java:10]: Yeah if you're using idea, it defaults to using asterisk for more than I think 2 or 3 imports.

Comment (LittleBigBug) [reply src/org/vivecraft/VSE.java]: Gotcha!

**Comment by Techjar** [comment]
Alright looks good, though can you squash the commits?

**Comment by LittleBigBug** [comment]
Rebased & squashed :)

**Comment by Techjar** [comment]
Somehow I don't think you did that right, It's even more of a mess now.

**Comment by LittleBigBug** [comment]
oh shit what happened, should I redo the PR?

**Comment by Techjar** [comment]
Nah at this point the easiest solution would be to just soft reset to `1.16` and then commit again.

**Comment by LittleBigBug** [comment]
I see, okay I will. Thank you. Also, I dont know why git keeps merging the branches which is why its messing up. Apologies for the mess :(

**Comment by Techjar** [comment]
Likely because you worked on the master branch, so now it's not so easy to rebase. If you can't figure it out it's fine, I'll just make the changes myself.

**Comment by LittleBigBug** [comment]
Finally, bout time. Squashed now

**Comment by Techjar** [comment]
Oh alright, looks good to me then. Dunno who's doing silly shit with internal nms types and why, but whatever. The fix will get included in the next release, which will probably end up being for 1.16.4.

**Review by Techjar — APPROVED** [review]

**Comment by LittleBigBug** [comment]
Well forks like yatopia aim to aggressively optimize stuff so I guess they have no boundaries for that!
Thank you :)

**Comment by Techjar** [comment]
Yeah I heard about this new fork, but it looked sketchy as there wasn't even any real info on the site.

**Comment by LittleBigBug** [comment]
They were renamed like 5 times, its just a fork hosted on github here: https://github.com/YatopiaMC/Yatopia
"yatopia" doesnt yield any search results for me but their old name "yapfa" brings it up.
And yeah, its pretty minuscule.. I imagine maybe they ran some benchmarks and one place somewhere using it this is a better option? Donno, regardless its dumb but yea, whatever.

**Pull request merged**